cmd/adapter/main.go

```diff
@@ -96,106 +177,91 @@ func run() error {
-			log.Warning("Subscriber close timed out")
 		}
 	}()
 
-	// Define event handler using the loaded adapter configuration and API client.
-	// Each message invocation receives its own context (ctx) from the broker.
-	// This ensures message isolation - cancellation or timeout of one message
-	// does not affect other messages.
-	handler := func(ctx context.Context, evt *event.Event) error {
-		// Safely render event data with nil check
-		dataStr := ""
-		if evt.Data() != nil {
-			dataStr = string(evt.Data())
-		}
-		log.Infof("Received event: id=%s type=%s source=%s data=%s", evt.ID(), evt.Type(), evt.Source(), dataStr)
-
-		// TODO: Process event using adapterConfig and apiClient
-		// Each API call MUST use ctx (the message context) for proper isolation:
-		//   resp, err := apiClient.Get(ctx, url)  // ctx ensures per-message timeout/cancellation
-		//
-		// 1. Extract params from event data using adapterConfig.Spec.Params
-		// 2. Execute preconditions using adapterConfig.Spec.Preconditions
-		//    - Make API calls using apiClient.Get(ctx, ...)/Post(ctx, ...)/etc.
-		//    - Extract response fields and evaluate conditions
-		// 3. Create/update Kubernetes resources using adapterConfig.Spec.Resources
-		// 4. Execute post actions using adapterConfig.Spec.Post.PostActions
-		//    - Report status back to HyperFleet API using apiClient
-
-		// Reference config and client to avoid unused variable warnings
-		_ = adapterConfig
-		_ = apiClient
-
-		log.Info("Event processed successfully")
-		return nil
-	}
-
-	// Subscribe and block until context is cancelled
-	// Let the broker consumer determine the topic to subscribe to from BROKER_TOPIC environment variable
-	if err := broker_consumer.Subscribe(ctx, subscriber, "", handler); err != nil {
-		// Context cancellation is expected during graceful shutdown, not an error
-		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
-			log.Info("Subscription stopped due to context cancellation")
-			// Not an error - graceful shutdown
-		} else {
-			// Actual error (e.g. connection failed)
-			log.Error(fmt.Sprintf("Subscription failed: %v", err))
-			return err
-		}
+	log.Info("Adapter started, waiting for events...")
+
+	// Wait for shutdown signal or fatal subscription error
+	select {
+	case <-ctx.Done():
+		log.Info("Context cancelled, shutting down...")
+	case err := <-fatalErrCh:
+		log.Error(fmt.Sprintf("Fatal subscription error, shutting down: %v", err))
+		cancel() // Cancel context to trigger graceful shutdown
 	}
 
-	log.Info("Waiting for graceful shutdown...")
+	// Close subscriber gracefully
+	log.Info("Closing broker subscriber...")
+	if err := subscriber.Close(); err != nil {
+		log.Error(fmt.Sprintf("Error closing subscriber: %v", err))
+	} else {
+		log.Info("Subscriber closed successfully")
+	}
 
 	// Give a small grace period for in-flight messages to complete
+	log.Info("Initiating graceful shutdown...")
 	time.Sleep(time.Second)
```

Comment:
I just noticed this, the time.second might not be enough for the subscriber to drain and we potentially lose messages. I would suggest something like 
```
shutdownCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
defer cancel()
if err := subscriber.CloseWithContext(shutdownCtx); err != nil {
    log.Error(...)
}
```
a little bit more graceful given context with timeout to allow the brokers queue to drain